### PR TITLE
Add Observer API

### PR DESCRIPTION
**Pull request by jmillan**

### CODE REVIEW

**Comment by jmillan** [comment]
NOTE: Doc is missing. On it.

File: src/Consumer.ts [hunk 1]
```diff
@@ -246,6 +269,9 @@ export class Consumer extends EnhancedEventEmitter
 
 		this._paused = false;
 		this._track.enabled = true;
+
+		// Emit observer event.
+		this._observer.safeEmit('resume');
```
Comment (ibc) [inline src/Consumer.ts:275]: What about `trackended` event?

File: src/Device.ts [hunk 2]
```diff
@@ -331,6 +334,16 @@ export class Device
 		return this._sctpCapabilities!;
 	}
 
+	/**
+	 * Observer.
+	 *
+	 * @emits newtransport - (transport: Transport)
```
Comment (ibc) [inline src/Device.ts:340]: There is no  `Transport` class but `SendTransport` and `RecvTransport `.

Comment (jmillan) [reply src/Device.ts:340]: correct

Comment (ibc) [reply src/Device.ts:340]: Forget about this, my fault.

**Review by ibc — APPROVED** [review]
LGTM, just a minor thing commented

File: src/Consumer.ts (hunk 1 above)
Comment (jmillan) [reply src/Consumer.ts:275]: done

**Pull request merged**